### PR TITLE
Correct annotations

### PR DESCRIPTION
**Pull request by ggaspersic**
Rewrote the configuration so that the binary actually uses the dependencies defined in the library. This allows us to have better overview where is what being used. 

Also corrected/reduced some imports and the usage of the `#[cfg(test)]` as in some cases they weren't correct/missing.

### CODE REVIEW

File: src/block_helpers.rs [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
-use crate::graph;
 use crate::optimizer::OptimizerSGD;
 use crate::port_buffer;
 use crate::regressor::{BlockCache, BlockTrait};
 use std::cmp::min;
 use std::mem;
 use std::slice;
 
+#[cfg(test)]
+use crate::graph;
```
Comment (SkBlaz) [inline src/block_helpers.rs:15]: Not sure I understand this one - is `graph` used only for tests?

File: src/block_normalize.rs [hunk 2]
```diff
@@ -266,6 +266,7 @@ impl BlockStopBackward {
 }
 
 /*
+#[cfg(test)]
```
Comment (SkBlaz) [inline src/block_normalize.rs:269]: If these are comments anyways, let's just remove

File: src/feature_transform_implementations.rs [hunk 3]
```diff
@@ -15,59 +13,63 @@ use crate::feature_transform_executor::{
...
-            NamespaceFormat::F32
-        );
-        Ok(Box::new(Self {
-            from_namespace: ExecutorFromNamespace {
-                namespace_descriptor: from_namespaces[0].namespace_descriptor,
-            },
-        }))
-    }
-}
+/// # Basic example of a "full blown" simple FunctionExecutorTrait
```
Comment (SkBlaz) [inline src/feature_transform_implementations.rs:16]: Remove if unused/dead code at this point

File: src/hogwild.rs [hunk 4]
```diff
@@ -89,14 +87,13 @@ impl HogwildWorker {
...
     pub fn train(&mut self, receiver: Arc<Mutex<Receiver<Vec<u32>>>>) {
-        let mut some_num = 0u64;
         loop {
             let buffer = match receiver.lock().unwrap().recv() {
                 Ok(feature_buffer) => feature_buffer,
-                Err(RecvError) => break, // channel was closed
+                Err(_) => break, // channel was closed
             };
             self.feature_buffer_translator
-                .translate(buffer.as_slice(), some_num);
```
Comment (SkBlaz) [inline src/hogwild.rs:99]: `some_num` = 0?

File: src/radix_tree.rs [hunk 5]
```diff
@@ -32,30 +30,19 @@ impl Default for RadixTreeNode {
...
-        }
-    }
 
     pub(crate) fn insert(&mut self, key: &[u8], value: NamespaceDescriptorWithHash) {
         let mut node = &mut self.root;
 
         for &byte in key {
             let child = &mut node.children[byte as usize];
-            node = child.get_or_insert_with(RadixTreeNode::new);
+            node = child.get_or_insert_with(RadixTreeNode::default);
```
Comment (SkBlaz) [inline src/radix_tree.rs:45]: nice!

**Review by SkBlaz — COMMENTED** [review]
:zap:

File: src/block_helpers.rs (hunk 1 above)
Comment (ggaspersic) [reply src/block_helpers.rs:15]: Yes, the `graph` is only used in the methods that are made & used in tests (slearn2, ssetup_cache2, ....). & this way it only imports it with when tests are run.

File: src/block_normalize.rs (hunk 2 above)
Comment (ggaspersic) [reply src/block_normalize.rs:269]: Removed.

File: src/feature_transform_implementations.rs (hunk 3 above)
Comment (ggaspersic) [reply src/feature_transform_implementations.rs:16]: Removed.

File: src/hogwild.rs (hunk 4 above)
Comment (ggaspersic) [reply src/hogwild.rs:99]: Line 92, yeap :)

**Review by SkBlaz — APPROVED** [review]

**Pull request merged**